cloudify/workflows/tasks.py

```diff
@@ -976,10 +975,11 @@ def dump(self):
         }
 
     def remote(self):
-        node_instance = get_node_instance(self._node_instance_id)
-        node_instance.state = self._state
-        update_node_instance(node_instance)
-        return node_instance
+        get_rest_client().node_instances.update(
+            self._node_instance_id,
+            force=True,
+            state=self._state
+        )
```

Comment:
Won't this result in an instance whose state is updated having its version reset to 1?

Reply:
nope! we never set the version field ourselves - it's actually a sqlalchemy built-in feature, and the version field is incremented with any update, https://github.com/cloudify-cosmo/cloudify-manager/blob/bd26be23cd7313584aa5175da4c4a1775410677f/rest-service/manager_rest/storage/resource_models.py#L1765-L1766

Reply:
you'll also notice that in the endpoint, the version from the request is used _only_ for the comparison: https://github.com/cloudify-cosmo/cloudify-manager/blob/bd26be23cd7313584aa5175da4c4a1775410677f/rest-service/manager_rest/rest/resources_v1/nodes.py#L210-L228

Reply:
OK- makes our interface feel a bit misleading but that's not something I think we should worry about as part of this.